projects/common/src/time/interval-duration.service.ts

```diff
@@ -45,7 +43,9 @@ export class IntervalDurationService {
     maximumDataPoints: number = 500
   ): TimeDuration[] {
     // Can make this configurable at some point, but for now, an interval musts produce at least 3 data points
-    return this.getAvailableIntervals(timeRange, 3, maximumDataPoints);
+    const availableIntervals = this.getAvailableIntervals(timeRange, 3, maximumDataPoints);
+
+    return availableIntervals.length === 0 ? [IntervalDurationService.MAX_SUPPORTED_INTERVAL] : availableIntervals;
```

Comment:
Yeah, i will have to put this check inside the `getAvailableInternals`. I moved it outside, because technically it can return an empty array and we were keeping this check 
```
public getAutoDurationFromTimeDurations(durations: TimeDuration[]): TimeDuration {
    if (durations.length === 0) {
    throw Error('No intervals supported at requested time range');
    
    }
```

We shouldn't be throwing any errors here. How do you suggest we handle the empty array case across function calls?

Reply:
> https://github.com/hypertrace/hypertrace-ui/pull/2081#discussion_r1140740733

You could leave it, since it would no longer be expected to be called with an empty array after fixing  `getAvailableIntervals`. Or continue to have it always default in that case, as it's doing in the current PR.

Reply:
updated